ext/search_kit_reports/Civi/SearchKitReports/ReportAfforms.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Civi\SearchKitReports;
+
+use Civi\Core\Event\GenericHookEvent;
+use Civi\Core\Service\AutoSubscriber;
+
+use CRM_SearchKitReports_ExtensionUtil as E;
+
+class ReportAfforms extends AutoSubscriber {
+
+  public static function getSubscribedEvents() {
+    return [
+      'civi.afform.get' => 'onGetAfforms',
+      // note: we want to hook in after the default "fallback" display
+      // has been provided - just to make small tweak
+      'civi.search.defaultDisplay' => ['alterDefaultSearchDisplay', -50],
+    ];
+  }
+
+  /**
+   * Enable toggle columns on the default search display for
+   * some searches used as Reports
+   *
+   * TODO: this is a bit convoluted - maybe we could just enable
+   * by default on *all* default search displays?
+   */
+  public function alterDefaultSearchDisplay(GenericHookEvent $e): void {
+    $savedSearchId = $e->savedSearch['id'];
+
+    if (!$savedSearchId) {
+      return;
+    }
+
+    $isTaggedReport = \Civi\Api4\EntityTag::get(FALSE)
+      ->addWhere('tag_id:name', '=', 'SearchKitReport')
+      ->addWhere('entity_id', '=', $savedSearchId)
+      ->addWhere('entity_table', '=', 'civicrm_saved_search')
+      ->execute()
+      ->first();
+
+    if ($isTaggedReport) {
+      $e->display['settings']['toggleColumns'] = TRUE;
+      return;
+    }
+
+  }
+
+  public function onGetAfforms(GenericHookEvent $e) : void {
+    $reportSearches = \Civi\Api4\SavedSearch::get(FALSE)
+      ->addWhere('tags:name', 'IN', 'SearchKitReport')
+      // templates are excluded by default otherwise
+      ->addWhere('is_template', 'IN', [TRUE, FALSE])
+      ->addSelect('name', 'label', 'api_params', 'base_module');
+
+    if ($e->getNames) {
+      // if we are searching for specific forms, check those that match the afreport
+      // prefix
+      $afreportNames = array_filter($e->getNames['name'] ?? [], fn($name) => \str_starts_with($name, 'afreport'));
+      if (!$afreportNames) {
+        // if no afreports we can return early
+        return;
+      }
+      // afreportMySearch => MySearch
+      $searchNames = array_map(fn ($afreportName) => \substr($afreportName, 8), $afreportNames);
+      $reportSearches->addWhere('name', 'IN', $searchNames);
+    }
+
+    foreach ($reportSearches->execute() as $search) {
+      $searchName = $search['name'];
+      $afformName = 'afreport' . $searchName;
+
+      // use the default display
+      $displayName = '';
```

Comment:
Potentially here we could check if a table type display has been created for the SavedSearch, and use that instead of the default display.

It feels like it gets a bit messy to me. you could be creating a display based on that SavedSearch for some other reason, and it will change your report without you realising.

This sort of thing makes me think it would be good to materialize the default display with an actual record, so you can edit the default display directly. (This could be a soft path to flattening the config by default - for most searches you would only ever need to use the default display, and then creating extra displays for the same search could be an "advanced UI" type feature)